osf/models/registrations.py

```diff
@@ -647,11 +899,101 @@ def reject(self, user):
+            kwargs.pop('old_subjects')
+        return super(DraftRegistration, self).save(*args, **kwargs)
+
+    def update(self, fields, auth=None, save=True):
+        """Update the draft registration with the given fields.
+
+        :param dict fields: Dictionary of field_name:value pairs.
+        :param Auth auth: Auth object for the user making the update.
+        :param bool save: Whether to save after updating the object.
+        """
+        if not fields:  # Bail out early if there are no fields to update
+            return False
+        for key, value in fields.items():
+            if key not in self.WRITABLE_WHITELIST:
+                continue
+            if key == 'title':
+                self.set_title(title=value, auth=auth, save=False)
+            elif key == 'description':
+                self.set_description(description=value, auth=auth, save=False)
+            elif key == 'category':
+                self.set_category(category=value, auth=auth, save=False)
+            elif key == 'node_license':
+                self.set_node_license(
+                    {
+                        'id': value.get('id'),
+                        'year': value.get('year'),
+                        'copyrightHolders': value.get('copyrightHolders') or value.get('copyright_holders', [])
+                    },
+                    auth,
+                    save=save
+                )
+        if save:
+            updated = self.get_dirty_fields()
+            self.save()
+
+        return updated
+
+
+class DraftRegistrationUserObjectPermission(UserObjectPermissionBase):
+    """
+    Direct Foreign Key Table for guardian - User models - we typically add object
+    perms directly to Django groups instead of users, so this will be used infrequently
+    """
+    content_object = models.ForeignKey(DraftRegistration, on_delete=models.CASCADE)
+
+
+class DraftRegistrationGroupObjectPermission(GroupObjectPermissionBase):
+    """
+    Direct Foreign Key Table for guardian - Group models. Makes permission checks faster.
+
+    This table gives a Django group a particular permission to a DraftRegistration.
+    For example, every time a draft reg is created, an admin, write, and read Django group
+    are created for the draft reg. The "write" group has write/read perms to the draft reg.
+
+    Those links are stored here:  content_object_id (draft_registration_id), group_id, permission_id
+    """
+    content_object = models.ForeignKey(DraftRegistration, on_delete=models.CASCADE)
+
+
+@receiver(post_save, sender='osf.DraftRegistration')
+def create_django_groups_for_draft_registration(sender, instance, created, **kwargs):
+    if created:
+        instance.update_group_permissions()
+        DraftRegistrationContributor.objects.get_or_create(
+            user=instance.initiator,
+            draft_registration=instance,
+            visible=True,
+        )
+        instance.add_permission(instance.initiator, ADMIN)
```

Comment:
When a DraftRegistration is created, need to create an admin/write/read Django group for each draft registration, and add the appropriate permissions to each group.  This puts the structure in place to then add users to different django groups to give them permissions to the DraftRegistration.

Draft Registration initiator automatically made an admin.